src/main/org/deidentifier/arx/aggregates/HierarchyBuilderDate.java

```diff
@@ -534,10 +534,11 @@ private String generalize(String input, Granularity granularity) {
         if (_range == null) {
             return sdf.format(date);
         } else {
-            int year = Integer.valueOf(sdf.format(date));
-            int lower = (year / _range) * _range;
-            int upper = ((year / _range) + 1) * _range;
-            return "[" + lower + ", " + upper + "[";
+            int dateUnit = Integer.valueOf(sdf.format(date));
+            int lower    = Integer.valueOf((dateUnit - 1) / (_range))  * (_range) + 1;
+            int upper    = lower + _range - 1;
+            String outputDate = "["+lower+"-"+upper+"]";

```

Comment:
There are blanks missing before and after "+" in this line.

Reply:
> There are blanks missing before and after "+" in this line.

Do you mean line 540? I kept the old line format. If you like I can update it as well. 

> have you checked wether this also works with Millenia, e.g.? Or are you very confident that the code in lines 537-540 is correct for all cases (not only quarters)?

I already tested it on the Month/Year, Quarter a year, Decade, Century, and Millennium cases and it seems to work.

Reply:
Yes, I mean line 540. And no: you did not keep the old line format ;)

![image](https://user-images.githubusercontent.com/4058833/145803031-684596cd-cd83-4c97-ba39-b16dfde6156c.png)

PS: Please don't close PRs when changes are requested, but push commit to your branch to adress them.